operator/pkg/client/client_factory.go

```diff
@@ -374,3 +385,22 @@ func setKubernetesDefaults(config *rest.Config) error {
 	}
 	return rest.SetKubernetesDefaults(config)
 }
+
+func (s *SingletonClients) checkAndResetMapper(obj runtime.Object) (*meta.RESTMapping, error) {
+	gvk := obj.GetObjectKind().GroupVersionKind()
+	mapping, err := s.discoveryShortcutExpander.RESTMapping(gvk.GroupKind(), gvk.Version)
+	if gvk.Empty() {
+		return mapping, nil
+	}
+	if err != nil {
+		if meta.IsNoMatchError(err) {
+			s.discoveryRESTMapper.Reset()
+			return nil, fmt.Errorf("resetting REST mapper to update resource mappings: %w", err)
+		}
+	}
+	return mapping, err
+}
+
+func (s *SingletonClients) ToRuntimeClient() client.Client {
```

Comment:
```suggestion
func (s *SingletonClients) Client() client.Client {
```

you do not know if its from the runtime or the control-plane in this struct. Also i would remove the `To` and remove it or replace it with `As` since youre just doing conversion

Reply:
This function is removed